app/api/Torrent.js

```diff
@@ -48,8 +48,21 @@ export default class Torrent {
 
     this.engine.add(magnetURI, { path: cacheLocation }, torrent => {
       const server = torrent.createServer();
-      server.listen(port);
-      this.server = server;
+      const thisRef = this;
+
+      detect(this.port).then((_port) => {
```

Comment:
Let's call `thisRef` and `_port` names that are more understandable to people that are new to the project. How about `resolvedPort` instead of `thisRef`? And maybe it would be good to add comments about what we're doing here.

One of the initiatives I'm trying to take is making making the codebase more legible to people that are new to the project.